### ISSUE TITLE
Extending security coding practices to vendor IP

### ISSUE THREAD

**Issue by martin-lueker** [lowRISC/opentitan#421]
Follow-on to #367, as brought up by #390.
Once we establish a set of RTL coding guidelines for OpenTitan, the question remains as how enforce this in external components.

**Comment by tjaychen** [comment]
@arnonsha 
@zi-v 
@sha-ron 

it would be great to hear your opinions here.

This is the same question @msfschaffner , @eunchan and I have been discussing. Ideally we would like to maintain a similar coding style and standard across all code, both open source and proprietary.

**Comment by zi-v** [comment]
IMO, guidelines like ‘Secure Hardware Design Guidelines’ (https://docs.opentitan.org/doc/security/implementation_guidelines/hardware/) should not be enforced for propriety IPs. Reasons:
- Large IP vendors may not be open to customizations of their IPs. So this may narrow down the licensing options (sometimes to none)
- Enforcing such guidelines on IP vendors may increase cost, risk and lead time. Note that IP vendors usually don't start from scratch and the cost/risk/lead may be proportional to the diff.
- Guidelines as above are generally good, but they leave much room for interpretation, tradeoffs and RoI considerations. It's not clear to me what exactly should to be enforced and how.
- The access to proprietary IP is limited (usually it is not in public domain), this limits the review capabilities and also the added value for the open community.

**Comment by tjaychen** [comment]
I understand your point Ziv.
But for the IPs that are Nuvoton developed and at least visible to Google, I feel like we need to at least come to a compromise on what we should expect. 

I think coding style we maybe can waive, but some of the security principles I feel would be similar.  Specifically regarding any single bit signals that are potentially vulnerable.

**Comment by zi-v** [comment]
I agree that security needs to be maintained, but there is room for judgement of which 'single bit signals', for example, should be further protected and how.
Maybe we should further discuss the details outside the public domain.

**Comment by tjaychen** [comment]
I'm adding this to the M3 milestone as it is something we have to conclude by Gold and is work in progress.
